### DISCUSSION TITLE
Additional info (geoip) for "New Device Logged In" email

### DISCUSSION THREAD

**Discussion by dionysius** [dani-garcia/vaultwarden#3103] [Ideas]
Feature request to add best-guessed geoip data to the IP output in the email.

I potentially will have non-technical users on my vaultwarden installation. They might have heard the Term "IP" and its an "address in the internet" and maybe so far as "that should be **my** address in the internet".

vaultwarden is a critical matter so the more we can present useful information to the average user the better, so they can make good decisions.

**Comment by dionysius** [comment]
I scraped through my mails what the different companies do:
- Spotify: country, time
- Some local company: No info, but the mail that someone just logged in
- Steam: IP, country
- LTT: OS, device, country
- Banking company A: City, country, IP, Device
- Banking company B: OS, device, IP, city, country

Excluded: many sites also require to accept the new device by auth code or link. Obviously the city information is often wrongly estimated, but the country was so far 100% success rate as far I remember.

**Comment by BlackDex** [comment]
I can add it to the meta list of feature request.
Depending on the used provider the details/accuracy could differ.
MaxMind is probably the first one that comes to mind. And it has support for more detailed files if people are willing to pay for it as far as i know.

**Comment by dionysius** [comment]
In my opinion country (or what free options offer) is a first good start.

- On alpine there is [geoip](https://pkgs.alpinelinux.org/package/edge/main/x86/geoip)
- On debian based distros there is [libgeoip1](https://packages.ubuntu.com/en/jammy/libgeoip1) (which seems to be MaxMind):
> Description: non-DNS IP-to-country resolver library
>  GeoIP is a C library that enables the user to find the country that any
>  IP address or hostname originates from. It uses a file based database.
>  .
>  This database simply contains IP blocks as keys, and countries as values and
>  it should be more complete and accurate than using reverse DNS lookups.
>  .
>  This package contains the shared library.

- Or parsing what `whois <ip>` is printing...

**Reply by BlackDex** [reply]
whois doesn't work everywhere. Also, it creates a dependency on an external tool which has very different return formats per region or IP range.
